pyvera/__init__.py

```diff
@@ -1195,7 +1204,7 @@ def get_pin_codes(self, refresh: bool = False) -> List[LockCode]:
             LOG.error("Got unsupported string %s: %s", val, ex)
 
         # Loop to create a list of codes
-        codes = []
+        codes = {}
```

Comment:
If slots are integers - why do you want to change this to a dictionary?

Reply:
I did not request this change, it should have been from the tag I forked from
I made chagne to the line 1219
`_, _, pin, name = code_addrs[2:6]`  added `:6` to the list index

Reply:
I guess you may need to rebase. 

But you’re branch currently includes this change which isn’t in the library.

Reply:
Sorry My bad, I did change code to be dict over list. 
I am formatting this more elegantly, over output string being just a CSV list.

Previously it was
` codes.append((slot, name, pin))` I changed to be `codes[slot]={'name': name, 'pin': pin}`

There is no dependency on hassio for this, this will be a new feature request there

Reply:
You changed codes from an array to a dict.

please put it back.

Reply:
Is the Change from array to dict not acceptable? Is it breaking anything else?

Reply:
You already know my views.

Reply:
Sorry not getting emotions over comments.
I would like to hear it if it is not acceptable and understand the reasoning why
I would love to discuss/hear about the same on other chat/messenger/meeting platforms if necessary

Reply:
As you already know I’m trying to keep this code stable.

I don’t currently see any good reason for changing `codes` from an array to a dict.

Reply:
Having the output as dict helps a ton on the hassio side or other applications that will have to parse a CSV otherwise. 
Looks like that is not good enough of an argument for you

Reply:
It doesn’t make any sense to me.

As far as I can see you’re creating a dictionary with a key of `slot` which is an integer.

Each value of the `codes` dictionary is itself a dictionary - and I agree these make sense because the keys are ‘ name` and `pin`.

I expect if you just change back the initial assignment of `codes` you code will work because you’re assigning into it with an array assignment anyway (I agree the assignment is better than the append)

So honestly it just looks like an mistake.

Reply:
Hope you agree to correct this mistake someday. 

Probably another PR to fight for. 

Am reverting to original for now.

Reply:
It’s not a mistake on my part.

Another comment like this I will stop reviewing and close this PR.